src/dune_rules/super_context.ml

```diff
@@ -10,6 +10,16 @@ let default_context_flags (ctx : Context.t) =
   let cxx =
     List.filter c ~f:(fun s -> not (String.is_prefix s ~prefix:"-std="))
   in
+  let c, cxx =
+    if
+      Lazy.force Ansi_color.stderr_supports_color
+      && Ocaml_config.ccomp_type ctx.ocaml_config <> Msvc
```

Comment:
I think this should use pattern matching to be future proof. Is there a way to make sure we're using gcc and not say clang?

Reply:
Note that the flag is compatible with both gcc and clang.
The type is below, maybe we can use the string to differentiate but I doubt it.
```ocaml
type t =
    | Msvc
    | Other of string
```

Reply:
I added some infrastructure to detect the C compiler in https://github.com/ocaml/dune/pull/3802 which is currently reviewed by @dra27

Reply:
I think we need a way to make sure that the compiler supports this (compared to making sure that the compiler is not msvc). Old versions of gcc and clang might not support it either. Do you know when this was added?

Reply:
Support is documented in GCC 4.9.0, released in April 2014.
Clang support was added in https://github.com/llvm/llvm-project/commit/7e2da79efaf866cd6a0142d154aea371714f3fea, in 2013 around llvm 3.3.

Apart from digging the history, another reliable way to tell if the flag is supported could be to test the C compiler supports it. Maybe using configurator?
As the flag is added in `:standard` it can also easily be deactivated, right?